src/components/FiltersPopover/FiltersPopover.tsx

```diff
@@ -0,0 +1,162 @@
+import clsx from 'clsx';
+import React, { useEffect, useRef } from 'react';
+import styles from './FiltersPopover.module.css';
+import Button from '../Button';
+import ButtonGroup from '../ButtonGroup';
+import Icon from '../Icon';
+import Popover, { PopoverProps } from '../Popover';
+
+export type FiltersPopoverProps = {
+  /**
+   * Form controls, form fields, or other relevant information that will be displayed in the filters popover.
+   */
+  children: React.ReactNode;
+  /**
+   * CSS class names that can be appended to the component.
+   */
+  className?: string;
+  /**
+   * CSS class names that can be appended to the footer button group.
+   */
+  footerButtonGroupClassName?: string;
+  /**
+   * Indicates status that filters have been selected, influencing toggle button variant.
+   */
+  hasSelectedFilters?: boolean;
+  /**
+   * Callback called when the clear button is called.
+   */
+  onClear?: () => void;
+  /**
+   * Callback called when filters drawer is closed.
+   */
+  onClose?: () => void;
+  /**
+   * Callback called when the apply button is called.
+   */
+  onApply?: () => void;
+  /**
+   * Popover placement options relative to the filters trigger button.
+   */
+  placement?: PopoverProps['placement'];
+  /**
+   * Text to be placed in the button that activates the Filters Popover
+   */
+  triggerText: string;
+};
+
+type FiltersPopoverRenderProps = {
+  /**
+   * Closes popover action render prop passed from Popover parent.
+   */
+  close: (
+    focusableElement?: HTMLElement | React.RefObject<HTMLElement>,
+  ) => void;
+  /**
+   * Popover open status render prop passed from Popover parent.
+   */
+  open: boolean;
+} & Omit<FiltersPopoverProps, 'placement'>;
+
+/**
+ * This helper component passes render props from <Popover> to handle onClose functionality
+ * that HeadlessUI Popover does not provide natively.
+ */
+const FiltersPopoverRender = ({
+  children,
+  className,
+  close,
+  footerButtonGroupClassName,
+  hasSelectedFilters,
+  onApply,
+  onClear,
+  onClose,
+  open,
+  triggerText,
+  ...other
+}: FiltersPopoverRenderProps) => {
+  /**
+   * Hooks to emulate an onClose callback for the Popover.
+   */
+  const firstRender = useRef(true);
```

Comment:
Could you help me explain the need for tracking the first render?

Reply:
Yea callback in `useEffect` always runs on first render, but the Popover renders as closed, causing `onClose` to be called on render. Tracking `firstRender` prevents this from happening

Reply:
Awesome. Could you add a comment about that too?

Reply:
Is there a way to check what the previous version of a prop was in functional react components? Because I think if I were doing this in the old component format, I would use `componentDidUpdate ` and check if 1) `open` is `false` and 2) `open` previously was `true` (so I know it did close as opposed to rendering as closed to start with).

Reply:
There [is](https://stackoverflow.com/questions/53446020/how-to-compare-oldvalues-and-newvalues-on-react-hooks-useeffect) but in the end its the same `useRef` `useEffect` combo